virttest/utils_secret.py

```diff
@@ -0,0 +1,52 @@
+"""
+Secret related utility functions
+"""
+
+import logging
+
+from virttest import virsh
+
+
+def get_secret_list(remote_virsh=None):
+    """
+    Get secret list by virsh secret-list from local or remote host.
+
+    :param remote_virsh: remote virsh shell session.
+    :return secret list
+    """
+    logging.info("Get secret list ...")
+    if remote_virsh:
+        secret_list_result = remote_virsh.secret_list()
+    else:
+        secret_list_result = virsh.secret_list()
+    secret_list = secret_list_result.stdout_text.strip().splitlines()
+    # First two lines contain table header followed by entries
+    # for each secret, such as:
+    #
+    # UUID                                  Usage
+    # --------------------------------------------------------------------------------
+    # b4e8f6d3-100c-4e71-9f91-069f89742273  ceph client.libvirt secret
+    secret_list = secret_list[2:]
```

Comment:
same as above, keep this inside `try..` too.

Reply:
updated

Reply:
updated

Reply:
As above and give the secret_list a default value []

Reply:
secret_list  will become [] even if secret_list[2:] is out of index
>>> secret_list=[0,1]
>>> secret_list = secret_list[2:]
>>> secret_list
[]
>>>